test/test_plugin_manager.rb

```diff
@@ -8,29 +8,141 @@ def with_no_gemfile
     FileUtils.mv "Gemfile.old", "Gemfile"
   end
 
-  def test_requiring_from_bundler
-    with_env("JEKYLL_NO_BUNDLER_REQUIRE", nil) do
-      assert Jekyll::PluginManager.require_from_bundler,
-             "require_from_bundler should return true."
-      assert ENV["JEKYLL_NO_BUNDLER_REQUIRE"], "Gemfile plugins were not required."
+  context "JEKYLL_NO_BUNDLER_REQUIRE set to `nil`" do
+    should "require from bundler" do
+      with_env("JEKYLL_NO_BUNDLER_REQUIRE", nil) do
+        assert Jekyll::PluginManager.require_from_bundler,
+               "require_from_bundler should return true."
+        assert ENV["JEKYLL_NO_BUNDLER_REQUIRE"], "Gemfile plugins were not required."
+      end
     end
   end
 
-  def test_blocking_requiring_from_bundler
-    with_env("JEKYLL_NO_BUNDLER_REQUIRE", "true") do
-      assert_equal false, Jekyll::PluginManager.require_from_bundler,
-                   "Gemfile plugins were required but shouldn't have been"
-      assert ENV["JEKYLL_NO_BUNDLER_REQUIRE"]
+  context "JEKYLL_NO_BUNDLER_REQUIRE set to `true`" do
+    should "not require from bundler" do
+      with_env("JEKYLL_NO_BUNDLER_REQUIRE", "true") do
+        assert_equal false, Jekyll::PluginManager.require_from_bundler,
+                     "Gemfile plugins were required but shouldn't have been"
+        assert ENV["JEKYLL_NO_BUNDLER_REQUIRE"]
+      end
     end
   end
 
-  def test_no_gemfile
-    with_env("JEKYLL_NO_BUNDLER_REQUIRE", nil) do
-      with_no_gemfile do
-        assert_equal false, Jekyll::PluginManager.require_from_bundler,
-                     "Gemfile plugins were required but shouldn't have been"
-        assert_nil ENV["JEKYLL_NO_BUNDLER_REQUIRE"]
+  context "JEKYLL_NO_BUNDLER_REQUIRE set to `nil` and no Gemfile present" do
+    should "not require from bundler" do
+      with_env("JEKYLL_NO_BUNDLER_REQUIRE", nil) do
+        with_no_gemfile do
+          assert_equal false, Jekyll::PluginManager.require_from_bundler,
+                       "Gemfile plugins were required but shouldn't have been"
+          assert_nil ENV["JEKYLL_NO_BUNDLER_REQUIRE"]
+        end
       end
     end
   end
+
+  context "require gems" do
+    should "check if configured gems are allowed" do
+      allow(Jekyll::External).to receive(:require_with_graceful_fail).and_return(nil)
+      site = double(:gems => %w(jemoji foobar))
+      plugin_manager = PluginManager.new(site)
+
+      expect(plugin_manager).to receive(:plugin_allowed?).with("jemoji")
+      expect(plugin_manager).to receive(:plugin_allowed?).with("foobar")
```

Comment:
Good point! I'll update this test to proof they are required instead of allowed.